src/gpu/nvidia/sycl_cuda_compat.hpp

```diff
@@ -35,11 +35,16 @@ T get_native_mem(const interop_handle &ih, U acc) {
             ih.get_native_mem<::sycl::backend::ext_oneapi_cuda>(acc));
 }
 
-template <typename T>
-void host_task(::sycl::handler &cgh, const T &task) {
-    cgh.host_task(task);
+template <typename HandlerT, typename FnT>
+void host_task(HandlerT &cgh, const FnT &task) {
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle ih) {
+#else
+    cgh.host_task([=](sycl::interop_handle ih) {
+#endif
+        task(std::move(ih));
+    });
```

Comment:
(nit) would it work for readability, to not split braces with macro conditionals and do something like:

```suggestion
#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
    cgh.ext_codeplay_enqueue_native_command(task);
#else
    cgh.host_task(task);
#endif
```

Reply:
Agreed, changes made.

Reply:
Do you need to wrap the task in another lambda compared to what Mourad suggested?

Reply:
Removed the unnecessary lambda, thanks for catching my miss @t4c1!